Shopify/cli
# Update the theme selector to no longer match partial theme names by d…

### WHY are these changes introduced?

Fixes https://github.com/Shopify/cli/issues/4320

### WHAT is this pull request doing?

This PR updates the theme selector code to avoid implicitly considering names that contain "each." While this behavior was desired for read-only scenarios (`shopify theme list`), it can lead to destructive behavior in other scenarios (`shopify theme push` updating the wrong theme).

### How to test your changes?

- Create themes with names that match.
- Notice the `shopify theme push -t <name>` only selects themes by using the exact name.
- Notice the `shopify theme list --name <partial>*` matches names using the `*`.

### Post-release steps

Update the [`shopify theme list`](https://shopify.dev/docs/api/shopify-cli/theme/theme-list) documentation to clarify the `*` pattern.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
